cdap-data-fabric/src/main/java/co/cask/cdap/spi/data/transaction/TransactionRunners.java

```diff
@@ -0,0 +1,254 @@
+   * @param runnable the {@link TxRunnable} to be executed inside a transaction
+   * @param <X> exception type of propagate type
+   * @throws X if failed to execute the given {@link co.cask.cdap.api.TxRunnable} in a transaction.
+   * If the TransactionFailureException has a cause in it, the cause is thrown as-is if it is an instance of X.
+   * @throws RuntimeException if cause is not an instance of X. The cause is wrapped with {@link RuntimeException}
+   * if it is not already a {@link RuntimeException}.
+   */
+  public static <X extends Throwable> void run(TransactionRunner txRunner,
+                                               TxRunnable runnable, Class<X> exception) throws X {
+    try {
+      txRunner.run(runnable);
+    } catch (TransactionException e) {
+      throw propagate(e, exception);
+    }
+  }
+
+  /**
+   * Executes the given {@link TxRunnable} using the given {@link TransactionRunner}.
+   *
+   * @param txRunner the {@link TransactionRunner} to use for txRunner execution.
+   * @param runnable the {@link TxRunnable} to be executed inside a transaction
+   * @param <X1> exception type of first propagate type
+   * @param <X2> exception type of second propagate type
+   * @throws X1 if failed to execute the given {@link TxRunnable} in a transaction. If the
+   * TransactionException has a cause in it, the cause is thrown as-is if it is an instance of X1.
+   * @throws X2 if failed to execute the given {@link TxRunnable} in a transaction. If the
+   * TransactionException has a cause in it, the cause is thrown as-is if it is an instance of X2.
+   * @throws RuntimeException if cause is not an instance of X1 or X2. The cause is wrapped with
+   * {@link RuntimeException} if it is not already a {@link RuntimeException}.
+   */
+  public static <X1 extends Throwable, X2 extends Throwable> void run(TransactionRunner txRunner,
+                                                                      TxRunnable runnable,
+                                                                      Class<X1> exception1,
+                                                                      Class<X2> exception2) throws X1, X2 {
+    try {
+      txRunner.run(runnable);
+    } catch (TransactionException e) {
+      throw propagate(e, exception1, exception2);
+    }
+  }
+
+  /**
+   * Executes the given {@link TxCallable} using the given {@link TransactionRunner}.
+   * @param txRunner the {@link TransactionRunner} to use for txRunner execution.
+   * @param callable the {@link TxCallable} to be executed inside a transaction
+   * @param <V> type of the result
+   * @return value returned by the given {@link TxCallable}.
+   * @throws  RuntimeException if failed to execute the given {@link TxRunnable} in a transaction.
+   * If the TransactionException has a cause in it, the cause is propagated.
+   */
+  public static <V> V run(TransactionRunner txRunner, TxCallable<V> callable) {
```

Comment:
What is the reason for not adding `run(TransactionRunner txRunner, TxRunnable<V> runnable)`?

Reply:
I missed that, will add